### PR TITLE
 T4939: VRRP startup-delay for bonding fix

### PR DESCRIPTION
**Pull request by fett0**
enabled when VyOS reboot /power-off a delay to avoid changing the state master when no-preenpt

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Migration from an old Vyatta component to vyos-1x, please link to related PR inside obsoleted component
- [ ] Other (please describe):

## Related Task(s)

* https://phabricator.vyos.net/T4939

## Component(s) name

vrrp
## How to test
<!---
Please describe in detail how you tested your changes. Include details of your testing
environment, and the tests you ran. When pasting configs, logs, shell output, backtraces,
and other large chunks of text, surround this text with triple backtics
```
like this
```
VyOS-CLI: 
```
vyos@vyos:~$ show configuration commands | match vrrp
set high-availability vrrp global-parameters startup-delay '300'

```
it should be present in globla-def : 

```

vyos@vyos:~$ show vrrp detail
------< Global definitions >------
 umask = 0177
 vrrp_startup_delay = 300

``


-->

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
<!--- The entire development process is outlined here: https://docs.vyos.io/en/latest/contributing/development.html -->
- [x] I have read the [**CONTRIBUTING**](https://github.com/vyos/vyos-1x/blob/current/CONTRIBUTING.md) document
- [x] I have linked this PR to one or more Phabricator Task(s)
- [x] I have run the components [**SMOKETESTS**](https://github.com/vyos/vyos-1x/tree/current/smoketest/scripts/cli) if applicable
- [x] My commit headlines contain a valid Task id
- [ ] My change requires a change to the documentation
- [ ] I have updated the documentation accordingly

### CODE REVIEW

File: interface-definitions/high-availability.xml.in [hunk 1]
```diff
@@ -11,6 +11,25 @@
...
         </properties>
         <children>
+          <node name="global-parameters">
+            <properties>
+              <help>VRRP Globas parameters</help>
+            </properties>
+            <children>
+              <leafNode name="startup-delay">
+                <properties>
+                  <help>Time vrrp startup process (in seconds)</help>
```
Comment (sever-sever) [inline interface-definitions/high-availability.xml.in]: VRRP should be in uppercase

File: smoketest/scripts/cli/test_ha_vrrp.py [hunk 2]
```diff
@@ -110,6 +112,10 @@ def test_02_simple_options(self):
             self.cli_set(group_base + ['authentication', 'type', 'plaintext-password'])
             self.cli_set(group_base + ['authentication', 'password', f'{group}'])
 
+            # Global parameters
+            config = getConfig(f'global_defs')
+            self.cli_set(global_param_base + ['startup-delay', f'{startup_delay}'])
```
Comment (sever-sever) [inline smoketest/scripts/cli/test_ha_vrrp.py:117]: We should check this parameter after commit if it present in the config

**Review by sever-sever — CHANGES_REQUESTED** [review]

File: interface-definitions/high-availability.xml.in (hunk 1 above)
Comment (fett0) [reply interface-definitions/high-availability.xml.in]: done

File: smoketest/scripts/cli/test_ha_vrrp.py (hunk 2 above)
Comment (fett0) [reply smoketest/scripts/cli/test_ha_vrrp.py:117]: added!

**Comment by sever-sever** [comment]
Something wrong with XML file name `data/templates/high-availability/<tagNode name="global-parameter">.xml`

**Review by sever-sever — APPROVED** [review]

File: data/templates/high-availability/keepalived.conf.j2 [hunk 3]
```diff
@@ -5,6 +5,11 @@
 global_defs {
     dynamic_interfaces
     script_user root
+{% if vrrp.global_parameters is vyos_defined %}
```
Comment (c-po) [inline data/templates/high-availability/keepalived.conf.j2]: This `if` is superfluous an can be dropped. Please remove it. Other then that all looks good.

**Review by c-po — CHANGES_REQUESTED** [review]

File: data/templates/high-availability/keepalived.conf.j2 (hunk 3 above)
Comment (fett0) [reply data/templates/high-availability/keepalived.conf.j2]: done!

**Pull request merged**